driver-core/src/main/java/com/datastax/driver/core/policies/ErrorAwarePolicy.java

```diff
@@ -0,0 +1,322 @@
+         * error rate threshold, this host's exclusion time will be reset to the start.
+         * <p/>
+         * Default value for the retry period is 5 seconds.
+         *
+         * @param retryPeriod the period to use for refreshing the ErrorTracker.
+         * @return this {@link Builder} instance, for method chaining.
+         */
+        public Builder withRetryPeriod(int retryPeriod) {
+            this.retryPeriod = retryPeriod;
+            return this;
+        }
+
+        /**
+         * Creates the {@link ErrorAwarePolicy} instance.
+         * @return the newly created {@link ErrorAwarePolicy}.
+         */
+        public ErrorAwarePolicy build() {
+            return new ErrorAwarePolicy(this);
+        }
+    }
+
+    class PerHostErrorTracker implements LatencyTracker {
+
+        private double errorsThreshold;
+
+        private ConcurrentMap<Host, Meter> hostsMeters = new ConcurrentHashMap<Host, Meter>();
+
+        private ConcurrentMap<Host, Long> exclusions = new ConcurrentHashMap<Host, Long>();
+
+        private ErrorsFilter errorsFilter;
+
+        PerHostErrorTracker(double errorsThreshold, ErrorsFilter errorsFilter) {
+            this.errorsThreshold = errorsThreshold;
+            this.errorsFilter = errorsFilter;
+        }
+
+        @Override
+        public void update(Host host, Statement statement, Exception exception, long newLatencyNanos) {
+            if (exception == null) {
+                return;
+            }
+            if (!errorsFilter.shouldConsiderError(exception, host)) {
+                return;
+            }
+            if (!hostsMeters.containsKey(host)) {
+                hostsMeters.putIfAbsent(host, new Meter());
+            }
+            Meter hostMeter = hostsMeters.get(host);
+            hostMeter.mark();
+            if (hostMeter.getOneMinuteRate() >= errorsThreshold) {
+                logger.debug(String.format("Host : %s has reached the maximum authorized number of errors. Its error rate is : %.3f. " +
+                                "It will now be ignored by the ErrorAware load balancing policy during a period of %d milliseconds.",
+                        host, hostMeter.getOneMinuteRate(), retryPeriodMillis));
+                // Replaces the previous value if there is one.
+                // Which accounts for when a host was previously excluded and has finished its retry period,
+                // AND when a host is excluded and its retry period not expired yet, but still crosses the threshold,
+                // then its retry period will be reset. Which in theory shouldn't happen much since, as soon as
+                // a host gets excluded, we shouldn't be sending queries to it so its rate shouldn't in theory cross
+                // the threshold again until the end of its retry period.
+                exclusions.put(host, System.currentTimeMillis());
```

Comment:
I agree, in this context system time makes sense.   Using a monotonic clock seems more important for accuracy of calculating rates, but not as important for this case (duration between inclusion checks)